### ISSUE TITLE
Hosted fields aren't getting the id set properly in all cases...

### ISSUE THREAD

**Issue by chadacious** [nathanstitt/react-braintree-fields#21]
After upgrading to your latest version, I was encountering problems where the HostedField wasn't getting the id attribute. When braintree tries to get the div element to load the iframe into, it couldn't find it because the id wasn't set. Classname was set properly, just not id.

So with the latest version I get:
![image](https://user-images.githubusercontent.com/30086553/77210739-f2ea6700-6abe-11ea-9ecc-45637737a193.png)

This is what it should look like:
![image](https://user-images.githubusercontent.com/30086553/77210613-abfc7180-6abe-11ea-8296-72a0cd3854cd.png)

The strange thing was that it indeed appears to be working in a different project I have. But I wasn't able to determine exactly what was different. I reverted to the previous react-braintree-fields version  for now.

**Comment by nathanstitt** [comment]
Thanks for the report @chadacious .  I'm guessing it's something to do with the recent change I made to no longer use `componentWillMount` internally since that's been deprecated and will be removed soon.

The strange thing is that it works fine here and on the demo at https://nathanstitt.github.io/react-braintree-fields/ 

Can you tell me what version of React you're using?  Maybe there's something incompatible with that?

**Comment by tusharr** [comment]
@nathanstitt I think this might be related. But I'm getting the following error in the console (via the `onError` prop on `Braintree`)

```
BraintreeError {name: "BraintreeError", code: "HOSTED_FIELDS_INVALID_FIELD_SELECTOR", message: "Selector does not reference a valid DOM node.", type: "MERCHANT", details: {…}}
name: "BraintreeError"
code: "HOSTED_FIELDS_INVALID_FIELD_SELECTOR"
message: "Selector does not reference a valid DOM node."
type: "MERCHANT"
details:
fieldSelector: "#braintree-field-wrapper-1"
fieldContainer: undefined
fieldKey: "number"
__proto__: Object
__proto__: Error
```

The fields do not get rendered at all in my case.

**Comment by nathanstitt** [comment]
@tusharr yes that does sound related.  I think I might see what the issue is, I'll make an update in a bit.

**Comment by nathanstitt** [comment]
I've just published version `1.5.0` which will hopefully correct the "field divs without ids" bug.  Please give it a try when you get a chance.

**Comment by chadacious** [comment]
I can confirm that it is working now. Thanks.
